sklearn/tests/test_common.py

```diff
@@ -268,3 +275,40 @@ def test_strict_mode_check_estimator():
 def test_strict_mode_parametrize_with_checks(estimator, check):
     # Ideally we should assert that the strict checks are Xfailed...
     check(estimator)
+
+
+def _generate_search_cv_instances():
+    for SearchCV, (Estimator, param_to_tune) in zip(
+        [GridSearchCV, RandomizedSearchCV],
+        [
+            (Ridge, {"alpha": [0.1, 1.0]}),
+            (LogisticRegression, {"C": [0.1, 1.0]}),
+        ],
+    ):
+        yield SearchCV(Estimator(), param_to_tune)
+
+    for SearchCV, (Estimator, param_to_tune) in zip(
+        [GridSearchCV, RandomizedSearchCV],
+        [
+            (Ridge, {"ridge__alpha": [0.1, 1.0]}),
+            (LogisticRegression, {"logisticregression__C": [0.1, 1.0]}),
+        ],
+    ):
+        yield SearchCV(make_pipeline(PCA(), Estimator()), param_to_tune)
```

Comment:
Do we need a test for checking a pipeline in a searchcv?

Reply:
Since we are making the instance ourself, I think it would be nice to have it as well.